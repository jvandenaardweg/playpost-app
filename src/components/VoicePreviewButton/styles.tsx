import {
  StyleSheet
} from 'react-native';
import colors from '../../constants/colors';

export default StyleSheet.create({
  container: {
    borderRadius: 30,
    height: 30,
    width: 30,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.black
  },
  icon: {
    position: 'relative',
    right: -1,
    color: colors.white
  }
});
