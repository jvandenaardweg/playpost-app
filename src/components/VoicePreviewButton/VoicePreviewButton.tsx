import React from 'react';
import { ActivityIndicator, TouchableOpacity } from 'react-native';

import * as Icon from '../../components/Icon';

import styles from './styles';

interface Props {
  isLoading?: boolean;
  isPlaying?: boolean;
  isActive?: boolean;
  onPress(): void;
}

export const VoicePreviewButton: React.FC<Props> = React.memo(({
  isLoading,
  isPlaying,
  isActive,
  onPress
}) => (
  <TouchableOpacity
    style={[styles.container, (isPlaying || isActive) ? styles.containerActive : null]}
    activeOpacity={1}
    onPress={onPress}
    hitSlop={{ top: 20, left: 20, right: 20, bottom: 20 }}>
    {!isPlaying && !isLoading && <Icon.FontAwesome5 name="play" size={9} style={styles.icon} />}
    {isPlaying && !isLoading && <Icon.FontAwesome5 name="pause" size={9} style={styles.icon} />}
    {isLoading && !isPlaying && <ActivityIndicator />}
  </TouchableOpacity>
));
