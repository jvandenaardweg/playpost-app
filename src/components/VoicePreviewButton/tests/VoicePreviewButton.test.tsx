import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { VoicePreviewButton } from '../VoicePreviewButton';

describe('VoicePreviewButton', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<VoicePreviewButton onPress={() => {}} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
