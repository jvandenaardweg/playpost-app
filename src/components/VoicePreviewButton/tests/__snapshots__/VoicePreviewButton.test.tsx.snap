// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoicePreviewButton rendering should render correctly 1`] = `
<View
  accessible={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C5C5C5",
      "borderRadius": 32,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "width": 32,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 9,
        },
        Object {
          "color": "#ffffff",
          "position": "relative",
          "right": -1,
        },
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "900",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
